src/factory.ts

```diff
@@ -1,30 +1,29 @@
-import { IComment, IIdentity, IReply } from './commentformat';
+import { ICellSelectionComment, IComment, IIdentity, IReply } from './commentformat';
 import { PartialJSONValue, UUID } from '@lumino/coreutils';
 import { getCommentTimeString } from './utils';
+import { Cell, ICellModel } from '@jupyterlab/cells';
 
-export interface ICommentFactory<T = any> {
-  createComment: (options: CommentFactory.ICommentOptions<T>) => IComment;
-  createCommentWithPrecomputedTarget: (
-    options: Exclude<ICommentOptions<T>, 'target'>,
-    target: PartialJSONValue
-  ) => IComment;
+// export interface ICommentFactory<T = any> {
+//   createComment: (options: CommentFactory.ICommentOptions<T>) => IComment;
+//   createCommentWithPrecomputedTarget: (
+//     options: Exclude<ICommentOptions<T>, 'target'>,
+//     target: PartialJSONValue
+//   ) => IComment;
 
-  readonly type: string;
-  readonly targetFactory: (target: T) => PartialJSONValue;
-}
+//   readonly type: string;
+//   readonly targetFactory: (target: T) => PartialJSONValue;
+// }
 
-/**
- * A class that creates comments of a given type.
- */
-export class CommentFactory<T = any> implements ICommentFactory<T> {
-  constructor(options: CommentFactory.IOptions<T>) {
-    const { type, targetFactory } = options;
+export abstract class ACommentFactory<T = any> {
 
+  // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
```

Comment:
We can probably just remove this as a rule from eslint, or just make it a warning.